plume-models/src/timeline/mod.rs

```diff
@@ -0,0 +1,845 @@
+
+            let gnu_post = Post::insert(
+                conn,
+                NewPost {
+                    blog_id: blogs[0].id,
+                    slug: "slug".to_string(),
+                    title: "About Linux".to_string(),
+                    content: SafeString::new("you must say GNU/Linux, not Linux!!!"),
+                    published: true,
+                    license: "GPL".to_string(),
+                    ap_url: "".to_string(),
+                    creation_date: None,
+                    subtitle: "".to_string(),
+                    source: "you must say GNU/Linux, not Linux!!!".to_string(),
+                    cover_id: None,
+                },
+                &r.searcher,
+            )
+            .unwrap();
+            gnu_post.update_tags(conn, vec![Tag::build_activity("free".to_owned()).unwrap()]).unwrap();
+            PostAuthor::insert(conn, NewPostAuthor {post_id: gnu_post.id, author_id: blogs[0].list_authors(conn).unwrap()[0].id}).unwrap();
+
+            unimplemented!();
+
+            Ok(())
+        });
+    }*/
+
+    #[test]
+    fn test_matches_keyword() {
+        let r = &rockets();
+        let conn = &r.conn;
+        conn.test_transaction::<_, (), _>(|| {
+            let (users, blogs) = blogTests::fill_database(conn);
+
+            let gnu_post = Post::insert(
+                conn,
+                NewPost {
+                    blog_id: blogs[0].id,
+                    slug: "slug".to_string(),
+                    title: "About Linux".to_string(),
+                    content: SafeString::new("you must say GNU/Linux, not Linux!!!"),
+                    published: true,
+                    license: "GPL".to_string(),
+                    ap_url: "".to_string(),
+                    creation_date: None,
+                    subtitle: "Stallman is our god".to_string(),
+                    source: "you must say GNU/Linux, not Linux!!!".to_string(),
+                    cover_id: None,
+                },
+                &r.searcher,
+            )
+            .unwrap();
+
+            let tl = Timeline::new_for_user(
+                conn,
+                users[0].id,
+                "Linux title".to_owned(),
+                "title contains Linux".to_owned(),
+            )
+            .unwrap();
+            assert!(tl.matches(r, &gnu_post, Kind::Original).unwrap());
+            tl.delete(conn).unwrap();
+            let tl = Timeline::new_for_user(
+                conn,
+                users[0].id,
+                "Microsoft title".to_owned(),
+                "title contains Microsoft".to_owned(),
+            )
+            .unwrap();
+            assert!(!tl.matches(r, &gnu_post, Kind::Original).unwrap());
+            tl.delete(conn).unwrap();
+
+            let tl = Timeline::new_for_user(
+                conn,
+                users[0].id,
+                "Linux subtitle".to_owned(),
+                "subtitle contains Stallman".to_owned(),
```

Comment:
(i'd rather it didn't… lolsob…)

Reply:
shall I edit that?

Reply:
We can do a cleanup PR and just remove all problematic techmen from our test examples